### PR TITLE
feat: deprecate setHeightByRows in favor of setAllRowsVisible in Grid

### PR DESCRIPTION
**Pull request by fhdhsni**
Fixes: #1009
related: [web-components PR 2099](https://github.com/vaadin/web-components/pull/2099)

- [X] Feature

## Checklist

- [X] The issue is created in the corresponding repository and I have referenced it.
- [X] I have added tests to ensure my change is effective and works as intended.
- [X] I have performed self-review and corrected misspellings.

### CODE REVIEW

File: vaadin-grid-flow-parent/vaadin-grid-flow/src/test/java/com/vaadin/flow/component/grid/GridTest.java [hunk 1]
```diff
@@ -61,4 +61,36 @@ public void selectItem_lazyDataSet_selectionWorks() {
         Assert.assertTrue(grid.getSelectedItems().contains("foo"));
     }
 
+    @Test
+    public void make_all_rows_visible_by_setHeightByRows() {
```
Comment (sissbruecker) [inline vaadin-grid-flow-parent/vaadin-grid-flow/src/test/java/com/vaadin/flow/component/grid/GridTest.java:65]: In Java you usually use camelCase for naming. The test naming convention is not super consistent in this repo, however at least in this class it seems to be `withPrecondition1_withPrecondition2_..._expectedResult`.

**Comment by sissbruecker** [comment]
@fhdhsni There are quite a few references to `setHeightByRows` in other Maven modules, at least in:
- vaadin-grid-flow-demo
- vaadin-grid-flow-integration-tests

It would make sense to update those too.

Regarding tests, there is one integration test for `setHeightByRows` specifically in `GridViewIT.heightByRows_allRowsAreFetched`, which tests the setup in `GridView.createHeightByRows`. We might want to duplicate that test to work with `setAllRowsVisible`.

**Comment by fhdhsni** [comment]
> @fhdhsni There are quite a few references to `setHeightByRows` in other Maven modules, at least in:
> 
> * vaadin-grid-flow-demo
> * vaadin-grid-flow-integration-tests
> 
> It would make sense to update those too.
> 
> Regarding tests, there is one integration test for `setHeightByRows` specifically in `GridViewIT.heightByRows_allRowsAreFetched`, which tests the setup in `GridView.createHeightByRows`. We might want to duplicate that test to work with `setAllRowsVisible`.

Regarding the IT, at first I wrote something similar for the new `setAllRowsVisible` but after a discussion with Tomi we decided that a unit test would suffice and there's no need for ITs in this case. The fact that `allRowsVisible` property is set to true is checked with the unit tests, and for the numbers of rows:
```
waitUntil(driver -> grid.getRowCount() == 50);
```
it's testing the underling web component which has its own tests in the WC repo.

File: vaadin-grid-flow-parent/vaadin-grid-flow/src/test/java/com/vaadin/flow/component/grid/GridTest.java (hunk 1 above)
Comment (fhdhsni) [reply vaadin-grid-flow-parent/vaadin-grid-flow/src/test/java/com/vaadin/flow/component/grid/GridTest.java]: done

**Review by tomivirkki — APPROVED** [review]

**Pull request merged**